app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt

```diff
@@ -867,6 +873,55 @@ class ComposeActivity :
         binding.addPollTextActionTextView.compoundDrawablesRelative[0].colorFilter = PorterDuffColorFilter(textColor, PorterDuff.Mode.SRC_IN)
     }
 
+    // Use to pass state from cropImage caller to result function
+    private var cropImageItemOld: QueuedMedia? = null
+
+    private val cropImage = registerForActivityResult(CropImageContract()) { result ->
+        val uriNew = result.uriContent
+        if (result.isSuccessful && uriNew != null) {
+            cropImageItemOld?.let { itemOld ->
+                val size = getMediaSize(getApplicationContext().getContentResolver(), uriNew)
+
+                lifecycleScope.launch {
+                    viewModel.addMediaToQueue(
+                        itemOld.type,
+                        uriNew,
+                        size,
+                        itemOld.description,
+                        itemOld
+                    )
+                }
+            }
+        } else if (result == CropImage.CancelledResult) {
+            Log.w("ComposeActivity", "Edit image cancelled by user")
+        } else {
+            Log.w("ComposeActivity", "Edit image failed: " + result.error)
+            displayTransientError(R.string.error_media_edit_failed)
+        }
+        cropImageItemOld = null
+    }
+
+    private fun editImageInQueue(item: QueuedMedia) {
+        // If input image is lossless, output image should be lossless.
+        // Currently the only supported lossless format is png.
+        val mimeType:String? = contentResolver.getType(item.uri)
+        val isPng:Boolean = mimeType != null && mimeType.endsWith("/png")
+        val context = getApplicationContext()
+        val tempFile = createNewImageFile(context, if (isPng) ".png" else ".jpg")
+
+        // "Authority" must be the same as the android:authorities string in AndroidManifest.xml
+        val uriNew = FileProvider.getUriForFile(context, BuildConfig.APPLICATION_ID + ".fileprovider", tempFile)
```

Comment:
The fancy method for getting a URI was necessary to get a content URI.
Without doing this, the mime type gets lost, and then we convert to JPEG if we ever re-crop the photo or even upload to the server.
I test this section with this code:
```
Log.w("ComposeActivity", "Input file (" + mimeType + ", " + (if (isPng) "(PNG)" else "") + "): " + item.uri + " Output file: " + uriNew)
```
I know the suggestion was to go PNG on transparent files specifically, but this will cover all transparent files and more besides (my phone makes screenshots as PNGs and I don't want those getting changed to JPEGs just because I cropped them).

Note this code will do the wrong thing on WEBP (these are sometimes lossy and sometimes lossless) but in my testing Tusky refused to attach those anyway. (GIFs get attached as videos so they don't matter either.)